### ISSUE TITLE
Can't stop inside builtin function's paramter functions.

### ISSUE THREAD

**Issue by panzhongxian** [mostafa/goja#17]
I'll describe the detail first. Also I have know why this occurred. But I have no idea how to fix it. Please help me. @mostafa 

I have tried three cases to make true this is a problem about builtins, rather than arrow-function or nested arrow-function.

In this case, there are 3 function calls:

1. line 2: `testClosure` created a `NewFunc`
2. line 5: `(s, v) => {...}` created a `NewArrowFunc`
3. line 3: `() => {...}` created a `NewArrowFunc`

The debugger can break on line 2 and line 3, but not on line 5.

```
  1
  2 function testClosure() {
  3   return (() => {
  4     const base = 10;
  5     return [ 1, 2, 3, 4, 5 ].reduce((s, v) => {
  6       s += v + base;
  7       {
  8         let x = 123;
  9         s += x;
 10         {
 11           let x = -123;
 12           s += x;
 13         }
 14       }
 15       return s;
 16     })
 17   })()
 18 }
 19
 20 testClosure()
```

The debug info seems like this(0xc00029f560  (s, v) => {...} prg is runned explicitly):

```
newFunc: {0xc00029f560  (s, v) => {...}
newFunc: {0xc00029f500  () => {...}

dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f3e0, dbg.vm.pc: 0, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f3e0, dbg.vm.pc: 1, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f3e0, dbg.vm.pc: 2, offset: 233, filename: test.js, line: 20
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f3e0, dbg.vm.pc: 3, offset: 244, filename: test.js, line: 20
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f440, dbg.vm.pc: 0, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f440, dbg.vm.pc: 1, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f440, dbg.vm.pc: 2, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f440, dbg.vm.pc: 3, offset: 223, filename: test.js, line: 17
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 0, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 1, offset: 0, filename: test.js, line: 1
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 2, offset: 52, filename: test.js, line: 4
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 3, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 4, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 5, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 6, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 7, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc000402c60, dbg.vm.prg: 0xc00029f500, dbg.vm.pc: 8, offset: 73, filename: test.js, line: 5
```

**Comment by panzhongxian** [comment]
I've read the goja source code to figure out what goja does when it runs `return [ 1, 2, 3, 4, 5 ].reduce((s, v) => {`:

1. call `typedArrayProto_reduce` to transfer the parameter as a `FunctionCall` with `toCallable`
2. call `func (f *arrowFuncObject) Call(call FunctionCall) Value`
3. call `func (f *baseJsFuncObject) _call(call FunctionCall, newTarget, this Value) Value`

Inside the `_call()` the stack grows and assign the FunctionCall's prg to vm. After that the `vm.run()` is called and we can't caputure any `breakpoint()` info there.

**Comment by panzhongxian** [comment]
I have push a debug branch in my repo <https://github.com/panzhongxian/goja/tree/debug_builtins>. You can invoke `go test` and will get the following debug info. As I anaylized before, there is no `breakpoint()` calls during the builtint call period:

```
...
dbg.vm: 0xc0002e8900, dbg.vm.prg: 0xc00002eea0, dbg.vm.pc: 15, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc0002e8900, dbg.vm.prg: 0xc00002eea0, dbg.vm.pc: 16, offset: 73, filename: test.js, line: 5
dbg.vm: 0xc0002e8900, dbg.vm.prg: 0xc00002eea0, dbg.vm.pc: 17, offset: 95, filename: test.js, line: 5
arrowFuncObject=
vm: 0xc0002e8900, vm.prg: 0x0, f.prg: 0xc00002ef00
arrowFuncObject=
vm: 0xc0002e8900, vm.prg: 0x0, f.prg: 0xc00002ef00
arrowFuncObject=
vm: 0xc0002e8900, vm.prg: 0x0, f.prg: 0xc00002ef00
arrowFuncObject=
vm: 0xc0002e8900, vm.prg: 0x0, f.prg: 0xc00002ef00
dbg.vm: 0xc0002e8900, dbg.vm.prg: 0xc00002eea0, dbg.vm.pc: 18, offset: 95, filename: test.js, line: 5
dbg.vm: 0xc0002e8900, dbg.vm.prg: 0xc00002ee40, dbg.vm.pc: 4, offset: 223, filename: test.js, line: 17
...
```

**Comment by mostafa** [comment]
@panzhongxian As you said [here](https://github.com/mostafa/goja/issues/17#issuecomment-1000047584):

> After that the `vm.run()` is called and we can't caputure any `breakpoint()` info there.

That's the point. You should check the old (and nearly obsolete) `vm.debugMode` or the new `vm.debugger` struct to see if you should run `vm.debug()` or `vm.run()`.

**Comment by panzhongxian** [comment]
Closed by #19
